pytradfri/type_hint.py

```diff
@@ -0,0 +1,105 @@
+"""Type hints for API responses."""
+from __future__ import annotations
+
+from typing import Any
+
+from pydantic import BaseModel, Field
+
+from .const import (
+    ATTR_AIR_PURIFIER_AIR_QUALITY,
+    ATTR_AIR_PURIFIER_CONTROLS_LOCKED,
+    ATTR_AIR_PURIFIER_FAN_SPEED,
+    ATTR_AIR_PURIFIER_FILTER_LIFETIME_REMAINING,
+    ATTR_AIR_PURIFIER_FILTER_LIFETIME_TOTAL,
+    ATTR_AIR_PURIFIER_FILTER_RUNTIME,
+    ATTR_AIR_PURIFIER_FILTER_STATUS,
+    ATTR_AIR_PURIFIER_LEDS_OFF,
+    ATTR_AIR_PURIFIER_MODE,
+    ATTR_AIR_PURIFIER_MOTOR_RUNTIME_TOTAL,
+    ATTR_APPLICATION_TYPE,
+    ATTR_BLIND_CURRENT_POSITION,
+    ATTR_CREATED_AT,
+    ATTR_DEVICE_BATTERY,
+    ATTR_DEVICE_FIRMWARE_VERSION,
+    ATTR_DEVICE_INFO,
+    ATTR_DEVICE_MANUFACTURER,
+    ATTR_DEVICE_MODEL_NUMBER,
+    ATTR_DEVICE_POWER_SOURCE,
+    ATTR_DEVICE_SERIAL,
+    ATTR_DEVICE_STATE,
+    ATTR_ID,
+    ATTR_LAST_SEEN,
+    ATTR_LIGHT_CONTROL,
+    ATTR_LIGHT_DIMMER,
+    ATTR_NAME,
+    ATTR_OTA_UPDATE_STATE,
+    ATTR_REACHABLE_STATE,
+    ATTR_START_BLINDS,
+    ATTR_SWITCH_PLUG,
+    ROOT_AIR_PURIFIER,
+    ROOT_SIGNAL_REPEATER,
+)
+
+
+class AirPurifierResponse(BaseModel):
```

Comment:
I'd move each response model to the module where its used. Eg move `AirPurifierResponse` to the `air_purifier.py` module.

Reply:
I moved them to one file as I was having problems with circular imports. That shouldn't be an issue with your proposal of inheriting from the resource class. I'll move them back.